libcontainerd/remote_unix.go

```diff
@@ -374,14 +377,18 @@ func (r *remote) runContainerdDaemon() error {
 	// Start a new instance
 	args := []string{
 		"-l", fmt.Sprintf("unix://%s", r.rpcAddr),
-		"--shim", "docker-containerd-shim",
 		"--metrics-interval=0",
 		"--start-timeout", "2m",
 		"--state-dir", filepath.Join(r.stateDir, containerdStateDir),
 	}
-	if r.runtime != "" {
-		args = append(args, "--runtime")
-		args = append(args, r.runtime)
+	if goruntime.GOOS == "solaris" {
+		args = append(args, "--shim", "containerd-shim", "--runtime", "runc")
```

Comment:
`runtime == runc`? I thought it was named `runz`? :)

Also this can be removed, if the user specify an invalid runtime this will fails and that's the expected behavior.